src/Native/Runtime/arm/Interlocked.S

```diff
@@ -13,17 +16,18 @@
 // r2 = comparand
 LEAF_ENTRY RhpLockCmpXchg32, _TEXT
 ALTERNATE_ENTRY RhpLockCmpXchg32AVLocation
-    bx      lr
+LOCAL_LABEL(CmpXchg32Retry):
+          ldrex        r3, [r0]
+          cmp          r2, r3
+          bne          LOCAL_LABEL(CmpXchg32Exit)
+          strex        r3, r1, [r0]
+          cbnz         r3, LOCAL_LABEL(CmpXchg32Retry)
+LOCAL_LABEL(CmpXchg32Exit):
+          bx           lr
 LEAF_END RhpLockCmpXchg32, _TEXT
 
-
-// WARNING: Code in EHHelpers.cpp makes assumptions about this helper, in particular:
-// - Function "InWriteBarrierHelper" assumes an AV due to passed in null pointer will happen at RhpLockCmpXchg64AVLocation
-// - Function "UnwindWriteBarrierToCaller" assumes the stack contains just the pushed return address
-// r0 = destination address
-// r1 = value
-// r2 = comparand
 LEAF_ENTRY RhpLockCmpXchg64, _TEXT
 ALTERNATE_ENTRY RhpLockCmpXchg64AVLocation
```

Comment:
Yes, this should be implemented for 32-bit platforms. It is implemented for CoreCLR in C++. You can look at what asm code is generated by the C++ compiler on CoreCLR for `COMInterlocked::CompareExchange64` to see how to implement it here. It is hard to implement it in C++ in CoreRT because of the null reference handling.